logstash.go

```diff
@@ -72,30 +109,40 @@ func (a *LogstashAdapter) Stream(logstream chan *router.Message) {
 		}
 
 		tags := GetContainerTags(m.Container, a)
+		fields := GetLogstashFields(m.Container, a)
 
 		var js []byte
 		var data map[string]interface{}
 
 		// Parse JSON-encoded m.Data
 		if err := json.Unmarshal([]byte(m.Data), &data); err != nil {
-			// The message is not in JSON, make a new JSON message.
-			msg := LogstashMessage{
-				Message: m.Data,
-				Docker:  dockerInfo,
-				Stream:  m.Source,
-				Tags:    tags,
+
+			msg := make(map[string]interface{})
```

Comment:
Now when you are no longer using LogstashMessage you could remove the type entirely and remove duplicate code for both setting the tags and marshalling the message. That would be a nice cleanup and also remove some indentations!